include/ccf/historical_queries_adapter.h

```diff
@@ -56,21 +75,35 @@ namespace ccf::historical
     ccf::SeqNo seqno,
     std::string& error_reason);
 
-  ccf::endpoints::EndpointFunction adapter_v3(
+  FMT_DEPRECATED ccf::endpoints::EndpointFunction adapter_v3(
     const HandleHistoricalQuery& f,
     ccf::AbstractNodeContext& node_context,
     const CheckHistoricalTxStatus& available,
     const TxIDExtractor& extractor = txid_from_header);
 
-  ccf::endpoints::ReadOnlyEndpointFunction read_only_adapter_v3(
+  FMT_DEPRECATED ccf::endpoints::ReadOnlyEndpointFunction read_only_adapter_v3(
     const HandleReadOnlyHistoricalQuery& f,
     ccf::AbstractNodeContext& node_context,
     const CheckHistoricalTxStatus& available,
     const ReadOnlyTxIDExtractor& extractor = txid_from_header);
 
-  ccf::endpoints::EndpointFunction read_write_adapter_v3(
+  FMT_DEPRECATED ccf::endpoints::EndpointFunction read_write_adapter_v3(
     const HandleReadWriteHistoricalQuery& f,
     ccf::AbstractNodeContext& node_context,
     const CheckHistoricalTxStatus& available,
     const TxIDExtractor& extractor = txid_from_header);
+
+  ccf::endpoints::ReadOnlyEndpointFunction read_only_adapter_v4(
+    const HandleReadOnlyHistoricalQuery& f,
+    ccf::AbstractNodeContext& node_context,
+    const CheckHistoricalTxStatus& available,
+    const CommandTxIDExtractor& extractor = txid_from_header,
+    const ErrorHandler& ehandler = default_error_handler);
+
+  ccf::endpoints::EndpointFunction read_write_adapter_v4(
+    const HandleReadWriteHistoricalQuery& f,
+    ccf::AbstractNodeContext& node_context,
+    const CheckHistoricalTxStatus& available,
+    const CommandTxIDExtractor& extractor = txid_from_header,
```

Comment:
```suggestion
    const ReadOnlyTxIDExtractor& extractor = txid_from_header,
    const ErrorHandler& ehandler = default_error_handler);

  ccf::endpoints::EndpointFunction read_write_adapter_v4(
    const HandleReadWriteHistoricalQuery& f,
    ccf::AbstractNodeContext& node_context,
    const CheckHistoricalTxStatus& available,
    const TxIDExtractor& extractor = txid_from_header,
```

`Commands` have no access to the KV, `ReadOnly`s have read access (via a `ReadOnlyTx` object), and unprefixed or `ReadWrite`s have read-write access (via a `Tx` object). It's possible that someone wants to read from the KV to decide the target TxID, and we also let them write to it if they really wanted to. We should retain that, and not force them to extract the TxID purely from the request (with no KV access) - our _default_ `txid_from_header` extractor happens to be command-only, and many will be command-only, but we don't want to lose the ability for apps to use the KV here.

Reply:
Done. Same for ErrorHandler, as discussed in person.